recursive_art.py

```
@@ -33,8 +56,25 @@ def evaluate_random_function(f, x, y):
         >>> evaluate_random_function(["y"],0.1,0.02)
         0.02
     """
-    # TODO: implement this
-    pass
+    if f[0] == 'prod':
+        return ((evaluate_random_function(f[1], x, y)) *
+                (evaluate_random_function(f[2], x, y)))
+    elif f[0] == 'avg':
+        return 0.5*(evaluate_random_function(f[1], x, y) +
+                    evaluate_random_function(f[2], x, y))
+    elif f[0] == 'absolute':
+        return abs(evaluate_random_function(f[1], x, y))
+    elif f[0] == 'crazy':
+        return ((2*evaluate_random_function(f[1], x, y) + 16) /
+                (3*evaluate_random_function(f[2], x, y) - 12))**3
+    elif f[0] == 'cos_pi':
+        return math.cos(math.pi * evaluate_random_function(f[1], x, y))
+    elif f[0] == 'sin_pi':
+        return math.sin(math.pi * evaluate_random_function(f[1], x, y))
+    elif f[0] == 'x':
+        return x
+    elif f[0] == 'y':
+        return y
 
```

You might want an additional `else:` here in case something breaks and you can return an error message or similar